### ISSUE TITLE
Species "Landing Page"

### ISSUE THREAD

**Issue by coffeejunk** [sharkipedia/sharkipedia#33]
A simple landing page for the individual species

![mockup](https://user-images.githubusercontent.com/205556/58512338-ddd8cc00-816a-11e9-8027-44c5b029d30c.jpg)

**Comment by sebpardo** [comment]
I think it would be neater to have this show the trait values for each species the same way Coral Traits does rather than having a table as the output. If people want the table, we can simply have the species-specific "Download csv" link at the bottom.

**Comment by creeas** [comment]
@sebpardo we went over multiple ideas yesterday. The issue with showing all trait values is that quite a few species will have multiple values for single traits. So which value do you display? We could default to the first value in the dataset, but then users might presume that is the only data. We could show a list of traits that have values, but this will not give an idea of what the trait values actually are.

**Comment by sebpardo** [comment]
I see that point, but presenting a table will have the same problem given you'd have to scroll down a lot to find the desired trait values, and this might be quite difficult to do if its a species for which there are lots of data.

I think the expandable observation count approach of Coral Traits is more desirable (e.g. <https://coraltraits.org/species/37>), where you can click show on the trait you're looking for and it gives you the information you're looking for in a very tidy way. Just my two cents.

**Comment by creeas** [comment]
Ah yeah...that observation counter is pretty cool. I had not noticed that before! So it shows the value if it is a single observation or shows the number of observations if there are multiple that can then be broken out. I think you are right @sebpardo ...this seems much better to have headings for each of the trait classes then similar simple table layout for trait names, observations (or number if multiple) and then a button to break out those values.

**Comment by sebpardo** [comment]
From our discussion today it seemed to me the ideal halfway point for the species landing page was the example image you have above but with separate tables for each trait class, so you can see the trait names and values, location, and reference (to begin with, might have to add a few extra columns down the line) for each measurement. This way, there is no nesting of pages (with show buttons) and all the relevant information can be seen, in a tidy way, from the species pages directly. 

Basically, instead of one big csv shown as output, it would be multiple smaller html tables, one for each trait class, showing all the measurements (i.e. rows) but only showing some key columns (trait name, trait value, location, reference).

**Comment by coffeejunk** [comment]
I've taken a first stab at this, see [here](https://shark-t-staging-pr-35.herokuapp.com/species)

<img width="1440" alt="Screen Shot 2019-06-03 at 10 13 22 PM" src="https://user-images.githubusercontent.com/205556/58846330-d030b400-864c-11e9-86fa-bba5041f62ce.png">

and [here](https://shark-t-staging-pr-35.herokuapp.com/species/970)
<img width="1438" alt="Screen Shot 2019-06-03 at 10 13 05 PM" src="https://user-images.githubusercontent.com/205556/58846318-c4dd8880-864c-11e9-8b77-b062140845b1.png">

let me know if you have any further thoughts as to which columns/values should be displayed

**Comment by sebpardo** [comment]
This is great! I'd only change two things:

- Show the full resource instead of the resource code, and do so at the rightmost column
- Add a column for standard right beside the value column

I honestly think this is all we need for the species search, and I imagine the trait search would be the same but flipped around.

**Comment by hkindsvater** [comment]
I agree with Seb, @NickDulvy what do you think? 

and thank you @coffeejunk for figuring this out so quickly.

**Comment by creeas** [comment]
Looks great @coffeejunk

**Comment by NickDulvy** [comment]
Looks great, concur with @sebpardo 
1. Column order should be Name, Value, ValueType, Location, Resource, with our resource code replaced with the column that has the full bibliographic ref.

**Comment by coffeejunk** [comment]
I've updated the column order and added Standard:

<img width="899" alt="Screen Shot 2019-06-04 at 2 02 14 PM" src="https://user-images.githubusercontent.com/205556/58902577-5c84ba80-86d1-11e9-908e-2c599c4dbf61.png">

That being said (and this just came up in the trends-call), I don't have any data for the `resource code` <> `full bibliographic ref` mapping (the traits data doesn't contain it). @creeas / @sebpardo do you have something I can use for this?

**Comment by creeas** [comment]
@sebpardo helped set up the reference database so he would know best how to get the full reference from the resource_id. But there should be a bibtex ref for every resource that can be easily compiled...

**Comment by sebpardo** [comment]
I'm not sure how (or if) the list of resource_ids was imported into the web app; the original table with the Shark-references scrape had a column with the full citation for each resource. I was of the impression that the whole table was already in the database, not just the list of resource_ids. This is something we definitely need to include so people using the app can see the resource behind each trait value.

The scraper is located at <https://github.com/alharry/shark-resources>, but it seems currently we don't have the output of the scrape as a CSV. I imagine it would make sense to load a lookup csv for resources when the db loads, just like we do for standards, methods, etc?

**Comment by coffeejunk** [comment]
@sebpardo all I have are the SharkTraits and Shark Trends Excel sheets and neither of them has the full references. If someone can generate such a CSV file for me that'd be very helpful.

**Comment by creeas** [comment]
I think the current ref list is saved the shark-resources as an .rds file `shark-recources.rds`. @coffeejunk I have generated a .csv file that is in the docs folder of the `restrictions` branch. Pushing now...

**Comment by coffeejunk** [comment]
Added a toggle for species list:

Species with data:
<img width="1414" alt="Screen Shot 2019-06-05 at 8 49 16 PM" src="https://user-images.githubusercontent.com/205556/58999352-6393f180-87d3-11e9-9cff-334e79e81c05.png">
All species:
<img width="1407" alt="Screen Shot 2019-06-05 at 8 49 44 PM" src="https://user-images.githubusercontent.com/205556/58999362-74446780-87d3-11e9-8c62-0e485dfc8f6a.png">

Added an overview page for the Traits:

<img width="1433" alt="Screenshot 2019-06-05 20 50 06" src="https://user-images.githubusercontent.com/205556/58999385-8b835500-87d3-11e9-8c6e-3edec7248e29.png">

And a detail page for the respective traits:
<img width="1151" alt="Screen Shot 2019-06-05 at 8 50 50 PM" src="https://user-images.githubusercontent.com/205556/58999402-9b9b3480-87d3-11e9-93c9-70daf27e31e8.png">


Let me know what you think and what other information you'd want on the Traits detail pages..

**Comment by hkindsvater** [comment]
I... think it is really looking good but am afraid I can't tell where this is deployed? Or are these previews only?

**Comment by coffeejunk** [comment]
You can view the current version here: https://shark-t-staging-pr-35.herokuapp.com/

**Comment by hkindsvater** [comment]
Thanks! looks great!

**Comment by coffeejunk** [comment]
Species landing page has been implemented and now includes a link to the data-export. 

<img width="1440" alt="Screen Shot 2019-07-24 at 5 31 22 PM" src="https://user-images.githubusercontent.com/205556/61806906-dcf5ab00-ae38-11e9-8bfe-023ef85855fc.png">
